lib/devise/passkeys/controllers/registrations_controller_concern.rb

```diff
@@ -208,15 +208,17 @@ def passkey_params
         end
 
         # @!visibility public
-        # Verifies that the `sign_up_params` has an `:email` and `:passkey_label`.
+        # Verifies that the `sign_up_params` has a value for the devise resource authentication
+        # key and `:passkey_label`.
         #
         # If either is missing or blank, a `400 Bad Request` JSON response is rendered.
         #
         # @example
         #  {"error": "Please enter your email address."}
+        #  {"error": "Please enter your username."}
         def require_email_and_passkey_label
```

Comment:
This method name and support to more than 1 auth key is the reason I've created this PR as draft and get your thoughts before going much further.

I could not think of a good name for this method and I also have not set this PR to support multiple auth keys.
But before proposing it, I would like to hear your thoughts on the approach to go!

Reply:
I think this should be left as-is, for the following reasons:
* Collecting the email address is largely used for:
  * Credential naming (which we're going to abstract using `default_passkey_name`)
  * Recovery flows
* Because passkeys are [discoverable credentials](https://developers.yubico.com/Passkeys/Passkey_concepts/Discoverable_vs_non-discoverable_credentials.html), we don't need to support `Devise.authentication_keys` in order for the authentication flows to work.
  * We can use `Devise.authentication_keys` to populate `default_passkey_name` (by default), but they are no longer required data as part of the authentication flow; they've become a labeling mechanism.
* Largely, recovery flows will require an email address to act as the destination for a recovery request. So the gem's default implementation should be to `require_email_and_passkey_label`
* If an application _does_ need to use an alternative recovery destination (such as a phone number), and they truly cannot use an email address; their application can `skip_before_action :require_email_and_passkey_label` and define their own registration `before_action` and `user_details_for_registration`, which they'll likely be doing in this case.

But definitely want to hear different opinions if someone doesn't agree! :)

Reply:
You make a good point about `email` being the most common mechanism for recovery, and the systems using other auth keys that is not email, most likely will have emails anyway...
It just feels like the name of the method is specific, while the implementation will be generic based and based on Devise.authentication_keys.

But I don't have a strong opinion on this and you also bring a valid and consistent point.
So... I am not the person you are looking for an opinion on this matter, as I agree with your point! :-D

Reply:
Hmmm, you're right about the two approaches contradicting each other: this controller `before_action` explicitly requires `email`, and `default_passkey_name` is implicitly `email` by default.

What if, instead, we made:

```ruby
def default_passkey_name
  self.email
end
```

Since:
* We're already operating under the presumption that an `email` will be present for the reasons outlined above
* Both paths can be easily overridden as we've outlined above & through this PR
* It makes it clear that `Devise.authentication_keys` are not used as part of `devise-passkeys`

Thoughts?

Reply:
There is still a valid scenario where email won't be present for registration for this user.
That is a scenario supported by Devise and devise-passkeys gem will force a valid Devise setup.

AWS Console for instance, apart from the root account you don't need email, and the auth key is account_id and username.

My comment [here](https://github.com/ruby-passkeys/devise-passkeys/pull/50/files#r1308181774) was about the label that shows up in the Passwords app, along with your passkey but was just a thinking out loud idea (I have a few others that maybe I should create in `Discussions` first?!?), and not to be implemented in the scope of this PR.

Reply:
> There is still a valid scenario where email won't be present for registration for this user.

Duh, you're right 🤦‍♂️. Given that, I don't think this change should be considered. Given the points outlined above about how the code can be easily overridden, it's better to keep the default functionality clear. I think this whole PR might need to be closed 😅 

> My comment [here](https://github.com/ruby-passkeys/devise-passkeys/pull/50/files#r1308181774) was about the label that shows up in the Passwords app, along with your passkey but was just a thinking out loud idea (I have a few others that maybe I should create in Discussions first?!?), and not to be implemented in the scope of this PR.

Yes, let's do that!